projects/packages/connection/src/class-external-storage.php

```diff
@@ -0,0 +1,237 @@
+<?php
+/**
+ * External Storage utilities for Jetpack Connection.
+ *
+ * Provides centralized logic for external storage implementations
+ * across different environments (WoA, VIP, other).
+ *
+ * Usage Example:
+ *
+ *     // 1. Create a storage provider class with required methods:
+ *     class My_Storage_Provider {
+ *         public function is_available() { return true; }
+ *         public function should_handle( $option_name ) {
+ *             return in_array( $option_name, array( 'blog_token', 'id' ), true );
+ *         }
+ *         public function get( $option_name ) {
+ *             // Return value from your external storage or null
+ *         }
+ *         public function get_environment_id() { return 'my_env'; } // Optional but recommended
+ *     }
+ *
+ *     // 2. Register the provider:
+ *     External_Storage::register_provider( new My_Storage_Provider() );
```

Comment:
Just to be on the safe side.
Somebody might implement it in a tool that runs with different versions of Jetpack.

```suggestion
 *     if ( class_exists( 'Automattic\Jetpack\Connection\External_Storage' ) ) {
 *         \Automattic\Jetpack\Connection\External_Storage::register_provider( new My_Storage_Provider() );
 *     }
```